gateway/service/src/main/java/org/eclipse/ditto/gateway/service/endpoints/routes/checkpermissions/CheckPermissionsResponse.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.ditto.gateway.service.endpoints.routes.checkpermissions;
+
+import static org.eclipse.ditto.base.model.common.ConditionChecker.checkNotNull;
+import static org.eclipse.ditto.base.model.signals.commands.CommandResponseHttpStatusValidator.validateHttpStatus;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import org.eclipse.ditto.base.api.common.CommonCommandResponse;
+import org.eclipse.ditto.base.model.common.HttpStatus;
+import org.eclipse.ditto.base.model.headers.DittoHeaders;
+import org.eclipse.ditto.base.model.json.JsonParsableCommandResponse;
+import org.eclipse.ditto.base.model.json.JsonSchemaVersion;
+import org.eclipse.ditto.base.model.signals.commands.CommandResponseJsonDeserializer;
+import org.eclipse.ditto.base.model.signals.commands.WithEntity;
+import org.eclipse.ditto.json.JsonField;
+import org.eclipse.ditto.json.JsonFieldDefinition;
+import org.eclipse.ditto.json.JsonObject;
+import org.eclipse.ditto.json.JsonObjectBuilder;
+import org.eclipse.ditto.json.JsonValue;
+
+/**
+ * Response on a {@link CheckPermissions} command.
+ *
+ * @since 3.7.0
+ */
+@Immutable
+@JsonParsableCommandResponse(type = CheckPermissionsResponse.TYPE)
+public final class CheckPermissionsResponse extends CommonCommandResponse<CheckPermissionsResponse> implements
+        WithEntity<CheckPermissionsResponse> {
+
+
+    /**
+     * The type of the response.
+     */
+    public static final String TYPE = TYPE_PREFIX + CheckPermissions.NAME;
+
+    private static final JsonFieldDefinition<JsonObject> JSON_PERMISSION_RESULTS =
+            JsonFieldDefinition.ofJsonObject("permissionResults");
+
+    private static final HttpStatus HTTP_STATUS = HttpStatus.OK;
+
+    private static final CommandResponseJsonDeserializer<CheckPermissionsResponse> JSON_DESERIALIZER =
```

Comment:
This is ununsed - "normally" such a `JSON_DESERIALIZER` is used in the `fromJson` method - maybe you wanted to do that?